metis-transformation/metis-transformation-service/src/main/java/eu/europeana/metis/transformation/service/XsltTransformer.java

```diff
@@ -32,37 +32,60 @@
 
   /**
    * Constructor in the case that no value of the datasetId field needs to be set.
-   * 
+   *
    * @param xsltUrl The URL of the XSLT file.
    * @throws TransformationException In case there was a problem setting up the transformation.
    */
   public XsltTransformer(String xsltUrl) throws TransformationException {
-    this(xsltUrl, null);
+    this(xsltUrl, null, null, null, null);
   }
 
   /**
    * Constructor.
-   * 
+   *
    * @param xsltUrl The URL of the XSLT file.
-   * @param datasetId The value that will be injected to the datasetId field in the XSLT. Can be
-   *        null.
+   * @param europeanaGeneratedIdsMap all the identifiers related to europeana RDF elements
+   * @param datasetName the dataset name related to the dataset
+   * @param edmCountry the Country related to the dataset
+   * @param edmLanguage the language related to the dataset
    * @throws TransformationException In case there was a problem with setting up the transformation.
    */
-  public XsltTransformer(String xsltUrl, String datasetId) throws TransformationException {
+  public XsltTransformer(String xsltUrl, EuropeanaGeneratedIdsMap europeanaGeneratedIdsMap,
```

Comment:
This class was designed to be created once for the transformation of multiple records, so that the (expensive) compilation of the XSLT could be done only once. The inclusion of the EuropeanaGeneratedIdsMap in the constructor here however forces a new instance to be created for every record. Maybe it is better to pass that object to the transform method instead (and set the parameter values there instead of in the constructor). 

Alternatively the object could be changed so that it can on-the-fly compute the parameters given the record ID, but still the values should then be set in the transform method, not in the constructor.